modules/bitgo/test/v2/unit/wallet.ts

```diff
@@ -1897,6 +1919,108 @@ describe('V2 Wallet:', function () {
+        });
+
+        txPrebuild.should.deepEqual({
+          walletId: tssPolygonWallet.id(),
+          wallet: tssPolygonWallet,
+          txRequestId: 'id',
+          txHex: 'ababcdcd',
+          buildParams: {
+            recipients,
+            type: 'transfertoken',
+          },
+          feeInfo: {
+            fee: 5000,
+            feeString: '5000',
+          },
+        });
+      });
+
+      it('should populate intent with EVM-like params', async function () {
+        const mpcUtils = new ECDSAUtils.EcdsaUtils(bitgo, bitgo.coin('tpolygon'));
+        // @ts-expect-error only pass in params being tested
+        const intent = mpcUtils.populateIntent(bitgo.coin('tpolygon'), {
+          intentType: 'tokenTransfer',
+          recipients,
+          feeOptions,
+        });
+        intent.should.have.property('feeOptions');
+        intent.feeOptions!.should.have.property('maxFeePerGas', 2000000000);
+        intent.feeOptions!.should.have.property('maxPriorityFeePerGas', 1000000000);
+        intent.should.have.property('recipients');
+        intent.recipients.should.have.property('length', 1);
+        intent.recipients[0].should.have.property('tokenData');
+        intent.recipients[0].tokenData!.should.have.property('value', recipients[0].tokenQuantity);
+        intent.recipients[0].tokenData!.should.have.property('type', recipients[0].type);
+        intent.recipients[0].tokenData!.should.have.property('symbol', recipients[0].tokenName);
+        intent.recipients[0].tokenData!.should.have.property('tokenContractAddress', recipients[0].tokenContractAddress);
+        intent.recipients[0].tokenData!.should.have.property('nftId', recipients[0].nftId);
+        intent.recipients[0].tokenData!.should.have.property('decimalPlaces', recipients[0].decimalPlaces);
+      });
+
+      it('should not populate intent with tokenData if certain params are undefined', async function () {
+        const mpcUtils = new ECDSAUtils.EcdsaUtils(bitgo, bitgo.coin('tpolygon'));
+        const recipients = [{
+          address: '0x101c3928946b2e1d99759e8e5d34b5e94c1a8e2f',
+          amount: '0',
+          tokenName: 'erc721:bitgoerc721',
+          tokenContractAddress: '0x8397b091514c1f7bebb9dea6ac267ea23b570605',
+          nftId: '38',
+          tokenQuantity: '1',
+          decimalPlaces: 0,
+        }];
+        // @ts-expect-error only pass in params be tested for
+        const intent = mpcUtils.populateIntent(bitgo.coin('tpolygon'), {
+          intentType: 'tokenTransfer',
+          recipients,
+          feeOptions,
+        });
+        intent.should.have.property('feeOptions');
+        intent.feeOptions!.should.have.property('maxFeePerGas', 2000000000);
+        intent.feeOptions!.should.have.property('maxPriorityFeePerGas', 1000000000);
+        intent.should.have.property('recipients');
+        intent.recipients.should.have.property('length', 1);
+        intent.recipients[0].should.not.have.property('tokenData');
```

Comment:
is it possible to throw if intent is transferToken and the needed params are not passed in